packages/shared/src/porter.ts

```diff
@@ -284,4 +330,88 @@ export class PorterClient {
     > = Object.fromEntries(decryptionResponses);
     return { encryptedResponses, errors };
   }
-}
+
+  public async sign191(
+    payload: Uint8Array,
+    cohortId: number,
+    options: SigningOptions = { optimistic: true, returnAggregated: true }
+  ): Promise<SignResult> {
+    const data: SignRequest = {
+      payload: toBase64(payload),
+      signature_type: 'eip191',
+      cohort_id: cohortId,
+      optimistic: options.optimistic,
+      return_aggregated: options.returnAggregated,
+    };
+
+    const resp: AxiosResponse<SignResponse> = await this.tryAndCall({
+      url: '/sign191',
+      method: 'post',
+      data,
+    });
+
+    const signingResults: { [ursulaAddress: string]: [string, string] } = {};
```

Comment:
You have to be careful here because errors can be returned in the signing result i.e. the Porter HTTP call result contains errors. See how `decrypt` is handled.

Similar comment for `signUserOp`.